### PR TITLE
Show a warning if the user attempts to leave a room that is invite only

### PR DESCRIPTION
**Pull request by Half-Shot**
Partial fix for
https://github.com/vector-im/riot-web/issues/2855
https://github.com/vector-im/riot-web/issues/5797

Signed-off-by: Will Hunt <will@half-shot.uk>

### CODE REVIEW

**Comment by t3chguy** [comment]
Welcome back :D

File: src/components/structures/MatrixChat.js [hunk 1]
```diff
@@ -847,16 +847,37 @@ export default React.createClass({
         }).close;
     },
 
+    _leaveRoomWarnings: function(roomId) {
+        const roomToLeave = MatrixClientPeg.get().getRoom(roomId);
+        // Show a warning if there are additional complications.
+        const roomState = roomToLeave.getLiveTimeline().getState("f");
```
Comment (t3chguy) [inline src/components/structures/MatrixChat.js]: ooi why this vs `roomToLeave.currentState` ?

Comment (t3chguy) [reply src/components/structures/MatrixChat.js]: oh RIP
> The state of the room at the time of the newest event in the timeline. Present for backwards compatibility - prefer getLiveTimeline().getState(false).

I've been using currentState for ages :L
--

Comment (t3chguy) [reply src/components/structures/MatrixChat.js]: turns out that doc is wrong because false !== 'f'

Comment (Half-Shot) [reply src/components/structures/MatrixChat.js]: screenshot from 2018-01-19 22-46-31

I'm guessing the docs are lying?

Comment (t3chguy) [reply src/components/structures/MatrixChat.js]: Yah thats why I said
> turns out that doc is wrong because false !== 'f'

 
I'll look into fixing that

Comment (Half-Shot) [reply src/components/structures/MatrixChat.js]: GH doesn't update comments quick enough. Anyway will fix indentation + using currentState instead.

Comment (t3chguy) [reply src/components/structures/MatrixChat.js]: Noooo, docs say prefer getState, the bit thats wrong is the arg
I'm lazy and use currentState, but if docs say prefer yours then you do you

**Comment by t3chguy** [comment]
Looks good to me, I'll throw it at someone

**Pull request merged**